Explorer/Assets/DCL/SDKComponents/Animator/Components/LoadedAnimation.cs

```diff
@@ -0,0 +1,76 @@
+using DCL.ECSComponents;
+using DCL.SDKComponents.Animator.Extensions;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace DCL.SDKComponents.Animator.Components
+{
+    public readonly struct LoadedAnimation
+    {
+        private readonly Animation animation;
+        private readonly Couple couple;
+
+        public LoadedAnimation(Animation animation)
+        {
+            this.animation = animation;
+            couple = new Couple();
+        }
+
+        public string? PlayingClip() =>
+            couple.PlayingAnimation?.Clip;
+
+        public void Initialize()
+        {
+            animation.Initialize();
+        }
+
+        public void Update()
+        {
+            if (this.couple is { HasPlayed: false, PlayingAnimation: { } })
+            {
+                animation.TryPlay(this.couple.PlayingAnimation.Clip!);
+                couple.HasPlayed = true;
+            }
+
+            if (this.couple.StoppedAnimation != null)
+                animation.TryStop(this.couple.StoppedAnimation.Clip!);
+        }
+
+        public void Apply(PBAnimationState? playingAnimation, PBAnimationState? stoppedAnimation)
+        {
+            this.couple.PlayingAnimation = playingAnimation;
+            this.couple.StoppedAnimation = stoppedAnimation;
+
+            if (playingAnimation != null)
+            {
+                animation.ApplySettings(playingAnimation);
+                couple.HasPlayed = false;
+            }
+
+            if (stoppedAnimation != null)
+                animation.ApplySettings(stoppedAnimation);
+        }
+
+        public static (PBAnimationState? playingAnimation, PBAnimationState? stoppedAnimation) RequiredAnimations(IEnumerable<PBAnimationState> animationStates)
+        {
+            PBAnimationState? playingAnimation = null;
+            PBAnimationState? stoppedAnimation = null;
+
+            foreach (PBAnimationState pbAnimationState in animationStates)
+                if (pbAnimationState.Playing)
+                    playingAnimation = pbAnimationState;
+                else if (pbAnimationState.ShouldReset)
+                    stoppedAnimation = pbAnimationState;
+
+            return (playingAnimation, stoppedAnimation);
+        }
+
+        private class Couple
+        {
+            public PBAnimationState? PlayingAnimation;
```

Comment:
Considering `PBAnimatorState` can be pooled and rented as a part of `PBAnimator`, I would not recommend to save it outside of `Protobuf` components. You can check if it is true, I can also assume it might be an edge case with `RepeatedField` so it's not pooled but I would not rely on that anyway.